src/desktop/mainwindow.cpp

```diff
@@ -98,23 +99,26 @@ static constexpr auto CTRL_KEY = Qt::CTRL;
 #include "libclient/drawdance/eventlog.h"
 #include "libclient/drawdance/perf.h"
 
-#include "desktop/dialogs/colordialog.h"
-#include "desktop/dialogs/newdialog.h"
-#include "desktop/dialogs/hostdialog.h"
-#include "desktop/dialogs/joindialog.h"
-#include "desktop/dialogs/layoutsdialog.h"
-#include "desktop/dialogs/logindialog.h"
-#include "desktop/dialogs/settingsdialog.h"
-#include "desktop/dialogs/resizedialog.h"
-#include "desktop/dialogs/playbackdialog.h"
-#include "desktop/dialogs/dumpplaybackdialog.h"
-#include "desktop/dialogs/flipbook.h"
-#include "desktop/dialogs/resetdialog.h"
-#include "desktop/dialogs/sessionsettings.h"
-#include "desktop/dialogs/serverlogdialog.h"
-#include "desktop/dialogs/tablettester.h"
-#include "desktop/dialogs/abusereport.h"
-#include "desktop/dialogs/brushsettingsdialog.h"
+#include "dialogs/colordialog.h"
```

Comment:
Includes should always be based from `src`. I feel like this is a losing battle because it is impossible to make compilers not implicitly include the current directory in the search path (I did think about creating a dummy directory containing every possible incorrect directory traversal to break code that doesn’t base includes from `src` though) but I am trying my best here to avoid the mess of confusing includes that existed previously where it was never clear where something was coming from.

Reply:
Ah, whoops, must have applied the wrong side of the patch. Yeah it was definitely confusing.

But not making the compiler include the file directory is very easy: use `<>` instead of `""`. Quotes will look relative to the current file's directory first and then fall back to the regular include path, angles will only look at the include path. That's also exactly how Drawdance uses them: quotes for relative includes, angles for everything else.

Weirdly, many projects, including IWYU and Drawpile, use angles only for external libraries and quotes for internal ones. Which, uh, is wrong, but happens to work because of the fallback behavior.